### PR TITLE
Feature/slack integration

### PR DESCRIPTION
**Pull request by dmitriy-uvin**
https://trello.com/c/1bAH35N4/378-add-slack-integration

### CODE REVIEW

File: api/.env.example [hunk 1]
```diff
@@ -91,3 +91,6 @@ FACEBOOK_REDIRECT=
 LINKEDIN_CLIENT_ID=
 LINKEDIN_CLIENT_SECRET=
 LINKEDIN_REDIRECT=
+
+SLACK_WEBHOOK_URL=
```
Comment (aLLeXUs) [inline api/.env.example:95]: Seems it is not used

File: api/app/Aggregates/Events/EventAggregate.php [hunk 2]
```diff
@@ -11,15 +11,15 @@ class EventAggregate
 {
     private int $id;
     private string $inviteeEmail;
-    private Carbon $startDate;
```
Comment (aLLeXUs) [inline api/app/Aggregates/Events/EventAggregate.php:14]: I am not sure the file should be changed here. @yevhen-turovets should be still working on this.

File: api/app/Mail/EventCreatedSlackToOwner.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+use App\Entity\Event;
+use App\Entity\EventType;
+use App\Entity\User;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Notifications\Messages\SlackMessage;
+use Illuminate\Queue\SerializesModels;
+
+class EventCreatedSlackToOwner extends SlackMessage implements ShouldQueue
```
Comment (aLLeXUs) [inline api/app/Mail/EventCreatedSlackToOwner.php:15]: It should be done properly. You do not need `getSlackMessage` method because the class is `SlackMessage` by itself. The class should be named as `EvantCreatedSlackMessage` and put to `app/Notifications/SlackMessages` folder.

**Review by aLLeXUs — CHANGES_REQUESTED** [review]
The work is not complete. The endpoints for adding and removing integration is missing. You should also conditionally send the notification (only if the integration is set up).

File: api/app/Mail/EventCreatedSlackToOwner.php (hunk 3 above)
Comment (dmitriy-uvin) [reply api/app/Mail/EventCreatedSlackToOwner.php:15]: How do you want it to be?

File: api/app/Aggregates/Events/EventAggregate.php (hunk 2 above)
Comment (yevhen-turovets) [reply api/app/Aggregates/Events/EventAggregate.php:14]: I change this line at last pull request

File: api/app/Mail/EventCreatedSlackToOwner.php (hunk 3 above)
Comment (aLLeXUs) [reply api/app/Mail/EventCreatedSlackToOwner.php]: Did you check `Illuminate\Notifications\Messages\SlackMessage` class sources? It is just a DTO. You can extend it and add the constructor that will receive the event and set the needed parameters, or you can create a static method that will do the same.

**Review by aLLeXUs — CHANGES_REQUESTED** [review]
`EventCreatedSlackMessage` changes needed. Also, it put into the wrong directory, has redundant traits and implementation (It needed for notification class, but not for this DTO).

File: api/app/Aggregates/Events/EventAggregate.php [hunk 4]
```diff
@@ -33,7 +33,7 @@ public function toArray(): array
             'id' => $this->id,
             'invitee_email' => $this->inviteeEmail,
             'start_date' => $this->startDate,
-            'event_type' => $this->eventType,
+            'event_type' => $this->eventTypeId,
```
Comment (aLLeXUs) [inline api/app/Aggregates/Events/EventAggregate.php:36]: Please remove the changes in the file to avoid merge conflicts. It is done in another PR.

File: api/app/Actions/Slack/ChangeActivitySlackNotificationsAction.php [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+use Illuminate\Support\Facades\Auth;
+
+final class ChangeActivitySlackNotificationsAction
+{
+    public function execute()
+    {
+        $user = Auth::user();
+
+        $user->slack_active = !$user->slack_active;
```
Comment (aLLeXUs) [inline api/app/Actions/Slack/ChangeActivitySlackNotificationsAction.php:15]: It is better not to switch but update from the parameter.

File: api/app/Actions/Slack/DeleteSlackNotificationsAction.php [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+use Illuminate\Support\Facades\Auth;
+
+final class DeleteSlackNotificationsAction
+{
+    public function execute()
+    {
+        $user = Auth::user();
+        $user->slack_webhook = null;
+        $user->slack_channel = null;
+        $user->slack_active = null;
```
Comment (aLLeXUs) [inline api/app/Actions/Slack/DeleteSlackNotificationsAction.php:16]: It will not update anything. You need to update these data via repository and then save.

**Review by aLLeXUs — APPROVED** [review]
